src/node.cc

```diff
@@ -2257,6 +2258,19 @@ inline napi_addon_register_func GetNapiInitializerCallback(DLib* dlib) {
       reinterpret_cast<napi_addon_register_func>(dlib->GetSymbolAddress(name));
 }
 
+// Initialize the addon, and close the DLib if this is a second-or-later
+// initialization call.
+#define CALL_ADDON_INIT(call, entry_point, dlib)              \
+  do {                                                        \
+    call;                                                     \
+    void* key_value = reinterpret_cast<void*>((entry_point)); \
+    if (addon_entry_points.count((key_value)) == 1) {         \
+      (dlib).Close();                                         \
+    } else {                                                  \
+      addon_entry_points.insert((key_value));                 \
+    }                                                         \
+  } while (0)
```

Comment:
Can this rather be an inline function?

Reply:
I'm not sure how to capture all the different ways we initialize an addon in a single inline function.

We have
* `callback(exports, module, context)` for well-known native addon symbols
* `napi_module_register_by_symbol(exports, module, context, napi_callback)` for N-API
* `mp->nm_context_register_func(exports, module, context, mp->nm_priv)` for self-registering context-aware addons
* `mp->nm_register_func(exports, module, mp->nm_priv)` for self-registering non-context-aware addons